### DISCUSSION TITLE
How to get pore network parameters?

### DISCUSSION THREAD

**Discussion by himanshu-sahu674** [PMEAL/porespy#771] [Q&A]
Hello Sir,
I have generated an image using 'rsa' generator and extracted pore network using snow2 and I want to get the pore network parameters like - 

1.coordination number - (By counting the number of pore bodies connected into a single pore, coordination number is obtained.)
2. Average pore size of image (pore radius in microns)
3. Average throat size of image (throat radius in microns)
4. Average throat length of image. 

For example I have generated an image by giving these parameters 

``` python

r1 = 60
r2 = 40
r3 = 30
r4 = 20
r5 = 10

n_max1 = 1 
n_max2 = 2
n_max3 = 5
n_max4 = 50
n_max5 = 100

c2 = -1
c3 = -2

im = ps.generators.rsa(im_or_shape=[400, 400], r = r1, n_max = n_max1, )
im = ps.generators.rsa(im_or_shape=im, r=r2,clearance = c2, n_max = n_max2,)
im = ps.generators.rsa(im_or_shape=im, r=r3,clearance = c3, n_max = n_max3,)
im = ps.generators.rsa(im_or_shape=im, r=r4, n_max = n_max4,smooth= False)
im = ps.generators.rsa(im_or_shape=im, r=r5,n_max= n_max5,return_spheres=False)

fig, ax = plt.subplots(1, 1, figsize=[4, 4])
ax.axis(False)
ax.imshow(im, origin='lower', interpolation='none');

```

and inverted the image where sphere represent the grains and also apply snow2 on inverted image to extract pore network

``` pytyhon
imt = ~ im
plt.imshow(imt)
```

this is the code that I have used to generate the image, so how can I get the above mentioned pore network parameters?

**Comment by jgostick** [comment]
Hi @himanshu-sahu674 
Once you've run ``snow2`` on the image you have to open it with OpenPNM to start interacting with it in the way that you want.  This is done by doing ``pn = op.io.network_from_porespy(sn.network)``, where ``sn`` is the name of the variable where you stored the result of the ``snow2`` extraction (like ``sn = ps.network.snow2(...)``).

Then in OpenPNM you do things like:

``plt.hist(pn['pore.inscribed_diameter'], bins=20)``

and

``plt.hist(pn.num_neighbors(pn.Ps))``

**Reply by himanshu-sahu674** [reply]
Sir,
After importing network and printing print(pn) I'm only getting these properties and labels
![Screenshot (900)](https://user-images.githubusercontent.com/85443615/205273578-a96c7e42-38bb-4867-90a9-d354f2fa2a1f.png)

while printing print(geo) I'm getting all these properties and also printed the hist plot for required properties
![Screenshot (901)](https://user-images.githubusercontent.com/85443615/205273817-009d0c44-4eee-458e-af2b-5666c1e48cc1.png)
![Screenshot (902)](https://user-images.githubusercontent.com/85443615/205273883-80e9cae3-d46a-4abd-908a-c3fbcd8decd3.png)

but again I don't know how to find coordination number.

**Reply by jgostick** [reply]
The coordination number is just the number of neighbors, so you can do ``pn.num_neighbors(pores=pn.Ps)``, which will return the number of neighbors for each pore.  You can then do a histogram of this result.
